examples/go-kafkacat/go-kafkacat.go

```diff
@@ -37,6 +42,63 @@ var (
 	sigs         chan os.Signal
 )
 
+var oauthbearerConfigRegex = regexp.MustCompile("^(\\s*(\\w+)\\s*=\\s*(\\w+))+\\s*$")
+var oauthbearerNameEqualsValueRegex = regexp.MustCompile("(\\w+)\\s*=\\s*(\\w+)")
+
+const (
+	saslDotOAuthBearerDotConfig = "sasl.oauthbearer.config"
+	principalClaimNameKey       = "principalClaimName"
+	principalKey                = "principal"
+	lifeSecondsKey              = "lifeSeconds"
+	joseHeaderEncoded           = "eyJhbGciOiJub25lIn0" // {"alg":"none"}
+)
+
+func retrieveUnsecuredToken(e kafka.OAuthBearerTokenRefresh, config *kafka.ConfigMap) (string, int64, string, map[string]string, error) {
+	v, err := config.Get(saslDotOAuthBearerDotConfig, "")
+	if err != nil {
+		return "", 0, "", nil, fmt.Errorf("ignoring event %T due to incorrect type for %s: %v", e, saslDotOAuthBearerDotConfig, err)
+	}
+	if !oauthbearerConfigRegex.MatchString(v.(string)) {
+		return "", 0, "", nil, fmt.Errorf("ignoring event %T due to malformed %s: %v", e, saslDotOAuthBearerDotConfig, v)
+	}
+	// set up initial map with default values
+	oauthbearerConfigMap := map[string]string{
+		principalClaimNameKey: "sub",
+		lifeSecondsKey:        "3600",
+	}
+	// parse the provided config and store name=value pairs in the map
+	for _, kv := range oauthbearerNameEqualsValueRegex.FindAllStringSubmatch(v.(string), -1) {
+		oauthbearerConfigMap[kv[1]] = kv[2]
+	}
+	principalClaimName := oauthbearerConfigMap[principalClaimNameKey]
+	mdPrincipal := oauthbearerConfigMap[principalKey]
+	lifeSeconds, lifeSecondsErr := strconv.Atoi(oauthbearerConfigMap[lifeSecondsKey])
+	// regexp is such that principalClaimName and lifeSeconds cannot end up blank,
+	// so check for a blank principal (which will happen if it isn't specified)
+	if mdPrincipal == "" {
+		return "", 0, "", nil, fmt.Errorf("ignoring event %T: no %s: %s=%s", e, principalKey, saslDotOAuthBearerDotConfig, v)
+	}
+	// sanity-check the provided lifeSeconds value, which must be a positive integer
+	if lifeSecondsErr != nil || lifeSeconds == 0 {
+		return "", 0, "", nil, fmt.Errorf("ignoring event %T: bad %s: %s=%s", e, lifeSecondsKey, saslDotOAuthBearerDotConfig, v)
+	}
+	// do not proceed if there are any unknown name=value pairs
+	if len(oauthbearerConfigMap) > 3 {
+		return "", 0, "", nil, fmt.Errorf("ignoring event %T: unrecognized key(s): %s=%s", e, saslDotOAuthBearerDotConfig, v)
+	}
+	now := time.Now().Unix()
+	mdLifetimeSeconds := now + int64(lifeSeconds)
+	oauthbearerMapForJSON := map[string]interface{}{
+		principalClaimName: mdPrincipal,
+		"iat":              now,
+		"exp":              mdLifetimeSeconds,
+	}
+	claimsJSON, _ := json.Marshal(oauthbearerMapForJSON)
+	encodedClaims := base64.RawURLEncoding.EncodeToString(claimsJSON)
+	jwsCompactSerialization := joseHeaderEncoded + "." + encodedClaims + "."
+	return jwsCompactSerialization, mdLifetimeSeconds, mdPrincipal, map[string]string{}, nil
```

Comment:
Maybe return a struct here? It's a bit difficult to understand what's being returned by looking at the method signature.

Reply:
👍 Added `type OAuthBearerToken struct {...}` in `handle.go`